ui/src/main.rs

```diff
@@ -104,362 +100,92 @@ pub static WINDOW_CMD_CH: Lazy<WindowManagerCmdChannels> = Lazy::new(|| {
-                        continue;
-                    }
-                    log::debug!("Loaded extension: {:?}", &path);
-                    extensions.insert(ext.details().meta.name.into(), ext);
-                }
-                Err(e) => {
-                    log::error!("Error loading extension: {:?}", e);
-                }
-            }
-        }
+fn app(cx: Scope) -> Element {
+    // 1. Make sure the warp engine is turned on before doing anything
+    bootstrap::use_warp_runner(cx);
 
-        Ok(())
+    // 2. Guard the app with the auth
+    let auth = use_state(cx, || AuthPages::Unlock);
+    let AuthPages::Success(identity) = auth.get() else {
+        return render! { AuthGuard { page: auth.clone() }};
     };
 
-    let user_extension_dir = fs::read_dir(&STATIC_ARGS.extensions_path)?;
-    add_to_extensions(user_extension_dir)?;
+    // 3. Make sure global context is setup before rendering anything downstream
+    bootstrap::use_boostrap(cx, identity)?;
 
-    if STATIC_ARGS.production_mode {
-        let uplink_extenions_path = common::get_extensions_dir()?;
-        let uplink_extensions_dir = fs::read_dir(uplink_extenions_path)?;
-        add_to_extensions(uplink_extensions_dir)?;
+    // 4. Throw up a loading screen until our assets are ready
+    if use_loaded_assets(cx).value().is_none() {
+        return render! { LoadingWash {} };
     }
 
-    Ok(extensions)
-}
-
-// called at the end of the auth flow
-#[inline_props]
-pub fn app_bootstrap(cx: Scope, identity: multipass::identity::Identity) -> Element {
-    log::trace!("rendering app_bootstrap");
-    let mut state = State::load();
-
-    if STATIC_ARGS.use_mock {
-        assert!(state.initialized);
-    } else {
-        state.set_own_identity(identity.clone().into());
+    // 5. Finally, render the app
+    render! {
+        Router::<UplinkRoute>{}
     }
+}
 
-    let desktop = use_window(cx);
-    // TODO: This overlay needs to be fixed in windows
-    if cfg!(not(target_os = "windows")) && state.configuration.general.enable_overlay {
-        let overlay_test = VirtualDom::new(OverlayDom);
-        let window = desktop.new_window(overlay_test, make_config());
-        state.ui.overlays.push(window);
-    }
-
-    let size = desktop.webview.inner_size();
-    #[cfg(target_os = "macos")]
-    let mut minimal_width = 600;
-
-    #[cfg(not(target_os = "macos"))]
-    let minimal_width = 600;
+// This needs to be in a layout since the notification listener needs a handle to the router
+// Eventually this restriction will be lifted once global contexts in dioxus are global accessible
+fn app_layout(cx: Scope) -> Element {
+    log::trace!("rendering app");
 
-    #[cfg(target_os = "macos")]
-    {
-        minimal_width = (minimal_width as f64 * desktop.webview.window().scale_factor()) as u32;
+    use_auto_updater(cx)?;
+    use_app_coroutines(cx)?;
+    use_router_notification_listener(cx)?;
+
+    render! {
+        AppStyle {}
```

Comment:
Similarly, the AppStyle component encapsulates a lot of the noise in the old component into a more narrow chunk. We're also more confident that we're not recomputing styles a lot since the component is memoized.